### PR TITLE
Use helm 3.5 with ct

### PR DESCRIPTION
**Pull request by joejulian**
**What type of PR is this?**

Chore/Feature

**What this PR does/ why we need it**:

Use helm 3.5 for testing. This includes, among other things, a more detailed linter that doesn't fail on api targets that will be deprecated in the future.

**Which issue(s) this PR fixes**:

**Special notes for your reviewer**:
6:30 is the best time on a clock, hands down.

**Does this PR introduce a user-facing change?**:

```release-note
none
```

**Checklist**

* [ ] *If a chart is changed, the chart version is correctly incremented.*
* [x] The commit message explains the changes and why are needed.
* [x] The code builds and passes lint/style checks locally.
* [x] The relevant subset of integration tests pass locally.
* [x] The core changes are covered by tests.
* [x] The documentation is updated where needed.

### CODE REVIEW

**Comment by joejulian** [comment]
This PR is only for testing. 3.5 won't be released until January 13.

**Comment by joejulian** [comment]
3.5 is released. rc2 was tested against all the charts and was tested against all incoming PRs that occurred since I opened this PR.

File: test/ct.yaml [hunk 1]
```diff
@@ -12,6 +12,10 @@ chart-repos:
   - kubefed-charts=https://raw.githubusercontent.com/kubernetes-sigs/kubefed/master/charts
   - jetstack-charts=https://charts.jetstack.io
   - banzaicloud=https://kubernetes-charts.banzaicloud.com
+<<<<<<< HEAD
```
Comment (dlipovetsky) [inline test/ct.yaml:15]: Unresolved merge conflict :smile:

File: test/helm3/ct-e2e.yaml [hunk 2]
```diff
@@ -34,6 +34,10 @@ chart-repos:
   - kubefed-charts=https://raw.githubusercontent.com/kubernetes-sigs/kubefed/master/charts
   - jetstack-charts=https://charts.jetstack.io
   - banzaicloud=https://kubernetes-charts.banzaicloud.com
+<<<<<<< HEAD
```
Comment (dlipovetsky) [inline test/helm3/ct-e2e.yaml:37]: Oops

File: test/helm3/ct.yaml [hunk 3]
```diff
@@ -12,6 +12,10 @@ chart-repos:
   - kubefed-charts=https://raw.githubusercontent.com/kubernetes-sigs/kubefed/master/charts
   - jetstack-charts=https://charts.jetstack.io
   - banzaicloud=https://kubernetes-charts.banzaicloud.com
+<<<<<<< HEAD
```
Comment (dlipovetsky) [inline test/helm3/ct.yaml:15]: Oops

Comment (joejulian) [reply test/helm3/ct.yaml]: dammit @juliangieseke ;)

I'm really not even sure why those are even there.

Comment (joejulian) [reply test/helm3/ct.yaml]: anyway.. thanks for catching it.

fixed.

**Comment by juliangieseke** [comment]
If it works w/o my commit, feel free to drop it. In my tests it didnt work, possibly because we didnt change the url in our codebase yet? 🤷

**Review by alejandroEsc — APPROVED** [review]
lgtm so far, though always get nervous about these changes

**Pull request merged**